lib/msal-node/src/network/HttpClient.ts

```diff
@@ -80,6 +87,7 @@ export class HttpClient implements INetworkModule {
                 url,
                 HttpMethod.POST,
                 options,
+                this.customAgent as https.Agent,
```

Comment:
How will this work with MSI? Calls to MSI are done over http.

Reply:
My improvements should address this.